satellite/satellitedb/satellitedbtest/cockroachnamespace.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedbtest
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/private/dbutil"
+	"storj.io/storj/private/dbutil/pgutil/pgtest"
+	"storj.io/storj/satellite"
+	"storj.io/storj/satellite/satellitedb"
+	dbx "storj.io/storj/satellite/satellitedb/dbx"
+)
+
+// NewCockroach creates a new satellite.DB that is used for testing. We create a new database with a
+// unique name so that there aren't conflicts when we run tests (since we may run the tests in parallel).
+// Postgres supports schemas for namespacing, but cockroachdb doesn't, so instead we use a different database for each test.
+func NewCockroach(log *zap.Logger, namespacedTestDB string) (satellite.DB, error) {
+	if err := CockroachDefined(); err != nil {
+		return nil, err
+	}
+
+	driver, source, err := dbutil.SplitConnstr(*pgtest.CrdbConnStr)
+	if err != nil {
+		return nil, err
+	}
+	db, err := dbx.Open(driver, source)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = db.Exec(fmt.Sprintf("CREATE DATABASE %s;", namespacedTestDB))
```

Comment:
Consider using instead:

    _, err = db.Exec(fmt.Sprintf("CREATE DATABASE %s;", pq.QuoteIdentifier(namespacedTestDB)))

That should make this work even when `namespacedTestDB` has unexpected special characters in it. We probably don't have to worry much about SQL injection here, but quoting everything is still often worthwhile.

(If you do want to make that change, do it also in Close() for the corresponding DROP)